simulation.cpp

```diff
@@ -408,20 +408,25 @@ void test_simulation() noexcept//!OCLINT test may be many
     }
 #endif
 
+//#define FIX_ISSUE_73
+#ifdef FIX_ISSUE_73
+
     ///Fitness of individuals is calculated based on how close they are to the current target value
     {
         int pop_size = 2;
         simulation s{0,0,pop_size};
 
+        env_param special_e_p;
+        special_e_p.env_function_A{optima_is_response_of_grn_first_ind};
+        environment special_env{special_e_p};
+
+        s.use_new_env(special_env);
```

Comment:
See the cheekyt test, unfortunately, as much as I like this design I see it difficult to implement unless we start using templates, and unfrotunately we would change the design of the architecture just for being able to implement the test this way. I think I found another way in #74 that does not require the implementation of templates, yet...

Reply:
Ah sry I already answered, anyway check this comment out for more in-depth explanatation.